Trakt/Configuration/PluginConfiguration.cs

```diff
@@ -46,8 +45,21 @@ public void AddUser(string userId)
     /// <param name="userId">The user id.</param>
     public void RemoveUser(string userId)
     {
-        var traktUsers = TraktUsers.ToList();
-        traktUsers.RemoveAll(user => user.LinkedMbUserId == userId);
-        TraktUsers = traktUsers.ToArray();
+        lock (_traktUsers)
+        {
+            _traktUsers.RemoveAll(user => user.LinkedMbUserId == userId);
+        }
+    }
+
+    /// <summary>
+    /// Gets a list of all trakt users.
+    /// </summary>
+    /// <returns>IReadonlyList{TraktUser} with all trakt users.</returns>
+    public IReadOnlyList<TraktUser> GetAllTraktUsers()
+    {
+        lock (_traktUsers)
+        {
+            return _traktUsers.AsReadOnly();
```

Comment:
Nothing really stops another thread from modifying the list after you return here.

Reply:
That's correct but I'm not sure how this influences the current code?

Reply:
Why have the lock at all?